homeassistant/components/nuimo_controller.py

```diff
@@ -0,0 +1,165 @@
+"""
+Component that connects to a Nuimo device over Bluetooth LE.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/nuimo_controller/
+"""
+import logging
+import threading
+import time
+from homeassistant.const import EVENT_HOMEASSISTANT_STOP
+
+REQUIREMENTS = [
+    '--only-binary=all '  # avoid compilation of gattlib
+    'git+https://github.com/getSenic/nuimo-linux-python'
+    '#nuimo==1.0.0&subdirectory=nuimo']
+
+DOMAIN = 'nuimo_controller'
+EVENT_NUIMO = 'nuimo_input'
+SERVICE_NUIMO = 'led_matrix'
+
+_LOGGER = logging.getLogger(__name__)
+_LOGGER.setLevel(logging.INFO)
+
+
+def setup(hass, config):
+    """Setup the Nuimo component."""
+    mac = config[DOMAIN].get('mac', None)
+    name = config[DOMAIN].get('name', None)
+    NuimoThread(hass, mac, name).start()
+    return True
+
+
+class NuimoLogger(object):  # pylint: disable=too-few-public-methods
+    """Handle Nuimo Controller event callbacks."""
+
+    def __init__(self, hass, name):
+        """Initialize Logger object."""
+        self._hass = hass
+        self._name = name
+
+    def received_gesture_event(self, event):
+        """Input Event received."""
+        _LOGGER.debug("received event: name=%s, gesture_id=%s,value=%s",
+                      event.name, event.gesture, event.value)
+        self._hass.bus.fire(EVENT_NUIMO,
+                            {'type': event.name, 'value': event.value,
+                             'name': str(self._name)})
+
+
+class NuimoThread(threading.Thread):
+    """Manage one Nuimo controller."""
+
+    def __init__(self, hass, mac, name):
+        """Initialize thread object."""
+        super(NuimoThread, self).__init__()
+        self._hass = hass
+        self._mac = mac
+        self._name = name
+        self._hass_is_running = True
+        self._nuimo = None
+        self._listener = hass.bus.listen_once(EVENT_HOMEASSISTANT_STOP,
+                                              self.stop)
+
+    def run(self):
+        """Setup connection or be idle."""
+        while self._hass_is_running:
+            if not self._nuimo or not self._nuimo.is_connected():
+                self._attach()
+                self._connect()
+            else:
+                time.sleep(1)
+
+        if self._nuimo:
+            self._nuimo.disconnect()
+            self._nuimo = None
+
+    def stop(self, event):  # pylint: disable=unused-argument
+        """Terminate Thread by unsetting flag."""
+        _LOGGER.debug('Stopping thread for Nuimo %s', self._mac)
+        self._hass_is_running = False
+        self._hass.bus.remove_listener(EVENT_HOMEASSISTANT_STOP,
+                                       self._listener)
+
+    def _attach(self):
+        """Create a nuimo object from mac address or discovery."""
+        # pylint: disable=import-error
+        from nuimo import NuimoController, NuimoDiscoveryManager
+        # logging.getLogger("nuimo").setLevel(logging.DEBUG)
```

Comment:
Remove commented code

If you are going to use logger, define global `_LOGGER = logging.getLogger(__name__)`